reporter/src/main/kotlin/reporters/WebAppReporter.kt

```diff
@@ -34,17 +35,17 @@ class WebAppReporter : Reporter {
         input: ReporterInput
     ) {
         val template = javaClass.classLoader.getResource("scan-report-template.html").readText()
-        val resultJson = jsonMapper.writeValueAsString(input.ortResult)
+        val evaluatedModel = EvaluatedModel.create(input)
 
-        val relevantResolutions = input.resolutionProvider.getResolutionsFor(input.ortResult)
-        val resolutionsJson = jsonMapper.writeValueAsString(relevantResolutions)
-
-        val result = template
-            .replace("id=\"ort-report-data\"><", "id=\"ort-report-data\">$resultJson<")
-            .replace("id=\"ort-report-resolution-data\"><", "id=\"ort-report-resolution-data\">$resolutionsJson<")
+        val prefix = template.substringBefore("id=\"ort-report-data\"><")
+        val suffix = template.substringAfter("id=\"ort-report-data\"><")
 
         outputStream.bufferedWriter().use {
-            it.write(result)
+            it.write(prefix)
+            it.write("id=\"ort-report-data\">")
+            evaluatedModel.toJson(it)
+            it.write("<")
+            it.write(suffix)
```

Comment:
How about if we would simply commit a comment like `<!-- Insert ORT report data here. -->` comment to `scan-report-template.html`, so the tag reads like

    id="ort-report-data"><!-- Insert ORT report data here. --></script>

and then do a simple

    template.replace("<!-- Insert ORT report data here. -->", evaluatedModel.toJson(it))

?

Reply:
I prefer we operate an on HTML element then a comment statement. My gut feeling is that working on a comment statement will be more errorprone.

Reply:
I'd say exactly the opposite is the case: The comment is a custom string that hardly will appear anywhere else in the file, but the code snippet might.

Reply:
The script element has an id which per HTML spec has to be unique throughout the document there is no such guarantee for code comments.

Reply:
> The script element has an id which per HTML spec has to be unique throughout the document

Well, throughout the *interpreted* part of the document, yes. But if someone would quote that code as part of a comment in the template, you'd be screwed. (Just as a hypothetical example.)

> there is no such guarantee for code comments.

Well, since you can pick the wording of the comment / marker yourself, I'd still regard it as safer.

But anyway, my real goal was to make the code doing the insertion of the data nicer. I'll come up with a new proposal that does not use a comment / marker string then.

Reply:
@sschuberth FYI, using the `replace` does not work, because for large scans the JSON string exceeds the JVM limit of 1GB for strings, that's why I have to write the JSON directly to the stream. I'll add a code comment about this separately after this is merged.

Reply:
Ok, but how about using something like this instead:

        val ortReportDataId = " id=\"ort-report-data\">"
        val index = template.indexOf(ortReportDataId)
        val prefix = template.substring(0, index)
        val suffix = template.substring(index, template.length)

        outputStream.bufferedWriter().use {
            it.write(prefix)
            it.write(ortReportDataId)
            evaluatedModel.toJson(it)
            it.write(suffix)
        }

This
- extracts a common string to a variable
- searches the `template` string only once
- gets rid of the lone `it.write("<")`

Reply:
I'll check how to improve this in a separate PR, probably with your suggestion.